### PR TITLE
Todo list Code Review Demo

### PR DESCRIPTION
**Pull request by mengiefen**
In this project, the following features has been implemented

- add todo to the list
- remove single todo from the list
- clear completed todos from the list
- update the todo description
- change the status of the todos
- saves the todo list array of objects in local storage.
- drag and drop option
- update the key in drag and drop operation

### CODE REVIEW

**Comment by omar25ahmed** [comment]
### **Hi @mengiefen, **
**Your project highlights:**
- Nice Application
- Creative UI design
- Complex Features working great
- ES6 is used in a very nice and organized way
- Your code fits the DRY JS best practices

Great job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### Required Changes:
Check the comments under the review.

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: src/modules/appendToDoList.js
Comment (omar25ahmed) [inline src/modules/appendToDoList.js]: - [x] Nice job so far here implementing adding the class tags dynamically but in my opinion, it would be better if you could these class tags to the array classNames by adding them to the same string that has the same order and index.

File: src/modules/app.js
Comment (omar25ahmed) [inline src/modules/app.js:51]: - [ ] That was very smart from you when you added this property also it will be better if you added `input.select()` so when the user clicks it will indicate him to edit the task.

File: src/modules/todoList.js
Comment (omar25ahmed) [inline src/modules/todoList.js:36]: https://www.awesomescreenshot.com/video/7517856?key=838c396a0da06127f5641b4ac6048a6e
- [x] There is a problem with the functionality here it's a simple issue but important when refreshing the page fo the users

**Review by omar25ahmed — CHANGES_REQUESTED** [review]

File: src/modules/app.js
Comment (mengiefen) [reply src/modules/app.js:51]: Dear @omar25ahmed,
I am not using input element to edit todo items. Therefore, I don't have `input.select()` method for paragraph elements. To include this feature I am using `contentEditable()` function to paragraph elements. I use some CSS properties to indicate the elements look like editable.

File: src/modules/todoList.js
Comment (mengiefen) [reply src/modules/todoList.js:36]: Thank you @omar25ahmed ,

I really appreciate you point out the lacking feature. I fixed this feature in this iteration.

File: src/modules/appendToDoList.js
Comment (mengiefen) [reply src/modules/appendToDoList.js]: Dear @omar25ahmed ,

I have changes these lines of code with `forEach()` method. Thank you for your suggestion.

**Comment by mengiefen** [comment]
Hi @omar25ahmed ,
I am really grateful for effort 💪🏽 and valuable suggestions. 🙇🏽‍♂️ 

I tried to included the suggested improvements. Please go through the submission and give me if I did as per your expectations. I need your approval to merge PR. If you have anything to add more as necessary changes, I will be thrilled and accept the changes. 

... waiting for your response!

**Pull request merged**
